pkg/kubelet/config/common.go

```diff
@@ -73,6 +73,21 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
 	// The generated UID is the hash of the file.
 	pod.Annotations[kubetypes.ConfigHashAnnotationKey] = string(pod.UID)
 
+	if isFile {
+		// Applying the default Taint tolerations to static pods,
+		// so they are not evicted when there are node problems.
```

Comment:
Good point. Probably the better thing to do here is

```
&api.Toleration{
   Operator: "Exists",
   Effect: api.TaintEffectNoExecute,
})
```

This will tolerate all NoExecute taints that are added in the future, which is presumably what we want for static pods.

I'm going to remove LGTM so @dchen1107 can make this change.